common/membership/resolver_test.go

```diff
@@ -1,114 +1,137 @@
+	a, _ := newTestResolver(t)
+	// add members to this ring
+	r, _ := a.getRing("test-worker")
+	ring := emptyHashring()
+	for _, addr := range []string{"127", "128"} {
+		host := NewHostInfo(addr)
+		ring.AddMembers(host)
+	}
 
-	logger := loggerimpl.NewNopLogger()
+	r.value.Store(ring)
 
-	rpm := testService.rings[0]
+	// the same ring will be picked here
+	hi, err := a.Lookup("test-worker", "key")
+	assert.Equal(t, "127", hi.GetAddress())
 
-	time.Sleep(time.Second)
+	// the same ring will be picked here
+	nohi, err := a.Lookup("WRONG-RING-NAME", "key")
+	assert.Error(t, err)
+	assert.Nil(t, nohi)
 
-	listenCh := make(chan *ChangedEvent, 5)
+	members, err := a.Members("test-worker")
+	assert.NoError(t, err)
+	assert.Equal(t, 2, len(members))
 
-	err := rpm.Subscribe("rpm-service-name", "test-listener", listenCh)
-	s.Nil(err, "AddListerener failed")
+	nomembers, err := a.Members("WRONG-RING-NAME")
+	assert.Error(t, err)
+	assert.Equal(t, 0, len(nomembers))
 
-	host, err := rpm.Lookup("rpm-service-name", "key")
-	s.Nil(err, "Ringpop monitor failed to find host for key")
-	s.NotNil(host, "Ringpop monitor returned a nil host")
+	memcount, err := a.MemberCount("test-worker")
+	assert.NoError(t, err)
+	assert.Equal(t, 2, memcount)
 
-	logger.Info("Killing host 1")
-	testService.KillHost(testService.hostUUIDs[1])
+	nomemcount, err := a.MemberCount("WRONG-RING-NAME")
+	assert.Error(t, err)
+	assert.Equal(t, 0, nomemcount)
 
-	select {
-	case e := <-listenCh:
-		s.Equal(1, len(e.HostsRemoved), "ringpop monitor event does not report the removed host")
-		s.Equal(testService.hostAddrs[1], e.HostsRemoved[0].GetAddress(), "ringpop monitor reported that a wrong host was removed")
-		s.Nil(e.HostsAdded, "Unexpected host reported to be added by ringpop monitor")
-		s.Nil(e.HostsUpdated, "Unexpected host reported to be updated by ringpop monitor")
-	case <-time.After(time.Minute):
-		s.Fail("Timed out waiting for failure to be detected by ringpop")
-	}
+	serr := a.Subscribe("test-worker", "sub1", changeCh)
+	assert.NoError(t, serr)
 
-	host, err = rpm.Lookup("rpm-service-name", "key")
-	s.Nil(err, "Ringpop monitor failed to find host for key")
-	s.NotEqual(testService.hostAddrs[1], host.GetAddress(), "Ringpop monitor assigned key to dead host")
+	serr = a.Subscribe("WRONG-RING-NAME", "sub1", changeCh)
+	assert.Error(t, serr)
 
-	err = rpm.Unsubscribe("rpm-service-name", "test-listener")
-	s.Nil(err, "Unsubscribe() failed")
+	serr = a.Unsubscribe("test-worker", "sub1")
+	assert.NoError(t, serr)
+
+	serr = a.Unsubscribe("WRONG-RING-NAME", "sub1")
+	assert.Error(t, serr)
 
-	rpm.Stop()
-	testService.Stop()
 }
 
-func (s *RpoSuite) TestCompareMembers() {
-	s.testCompareMembers([]string{}, []string{"a"}, true)
-	s.testCompareMembers([]string{}, []string{"a", "b"}, true)
-	s.testCompareMembers([]string{"a"}, []string{"a", "b"}, true)
-	s.testCompareMembers([]string{}, []string{"a"}, true)
-	s.testCompareMembers([]string{}, []string{"a", "b"}, true)
```

Comment:
ah, the duplicates came from here.  maybe check these lines' history, see if there's some failed refactor?